src/Compilers/CSharp/Portable/Binder/ForLoopBinder.cs

```diff
@@ -27,21 +27,28 @@ override protected ImmutableArray<LocalSymbol> BuildLocals()
             var declaration = _syntax.Declaration;
             if (declaration != null)
             {
-                var refKind = _syntax.RefKeyword.Kind().GetRefKind();
-
-                foreach (var variable in declaration.Variables)
+                if (declaration.Deconstruction == null)
                 {
-                    var localSymbol = MakeLocal(refKind,
-                                                declaration,
-                                                variable,
-                                                LocalDeclarationKind.ForInitializerVariable);
-                    locals.Add(localSymbol);
+                    var refKind = _syntax.RefKeyword.Kind().GetRefKind();
 
-                    if (variable.Initializer != null)
+                    foreach (var variable in declaration.Variables)
                     {
-                        PatternVariableFinder.FindPatternVariables(this, locals, variable.Initializer.Value);
+                        var localSymbol = MakeLocal(refKind,
+                                                    declaration,
+                                                    variable,
+                                                    LocalDeclarationKind.ForInitializerVariable);
+                        locals.Add(localSymbol);
+
+                        if (variable.Initializer != null)
+                        {
+                            PatternVariableFinder.FindPatternVariables(this, locals, variable.Initializer.Value);
```

Comment:
`PatternVariableFinder` was introduced to support pattern-matching, but was generalized by @AlekseyTs to handle other constructs such as `out` variables. I'll write an issue for this point.